### ISSUE TITLE
Review docker environment

### ISSUE THREAD

**Issue by tschaffter** [tschaffter/debian#7]
- Changing the ownership of `/var/run/docker.sock` is not ideal; check if it's still required after updating to WLS2 and Docker Desktop

**Comment by tschaffter** [comment]
In `docker-compose.yml`, the following was working with WSL 1:

```
    command: >
      bash -c "sudo chown root:docker /var/run/docker.sock
      && /bin/bash"
```

The issue with WSL 2 (possibly), is that the group `docker` in the container is matched to `999` in WSL. It then no longer possible to start a new container that mount `/var/run/docker.sock`.

```terminal
tschaffter@DESKTOP-H29C5A0:~/dev/debian$ ls /mnt/wsl/docker-desktop/shared-sockets/guest-services/docker.sock
/mnt/wsl/docker-desktop/shared-sockets/guest-services/docker.sock
tschaffter@DESKTOP-H29C5A0:~/dev/debian$ ls -al /mnt/wsl/docker-desktop/shared-sockets/guest-services/docker.sock
srw-rw---- 1 root root 0 Jul 31 21:25 /mnt/wsl/docker-desktop/shared-sockets/guest-services/docker.sock
tschaffter@DESKTOP-H29C5A0:~/dev/debian$ ls -al /var/run/docker.sock
srw-rw---- 1 root docker 0 Jul 31 21:25 /var/run/docker.sock
tschaffter@DESKTOP-H29C5A0:~/dev/debian$ ./debian.sh docker local
Pulling docker ... done
tschaffter@docker:~$ docker ps
CONTAINER ID        IMAGE                            COMMAND                  CREATED             STATUS              PORTS               NAMES
9d35fb52e7c3        tschaffter/debian:docker-local   "bash -c 'sudo chown…"   1 second ago        Up 1 second                             debian_docker_run_e949e96d9589
tschaffter@docker:~$ ^C
tschaffter@docker:~$ exit
tschaffter@DESKTOP-H29C5A0:~/dev/debian$ ./debian.sh docker local
ERROR: Couldn't connect to Docker daemon at http+docker://localhost - is it running?

If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.
tschaffter@DESKTOP-H29C5A0:~/dev/debian$ ls -al /var/run/docker.sock
```

**Comment by tschaffter** [comment]
## Update

I can successfully pass `/var/run/docker.sock` to a docker container run from WSL2 and user docker using `sudo docker <command>`. Using `docker <command>` without sudo in the container does not work so far (see below).

By default in WSL, `/var/run/docker.sock` has the ownership set to `root:docker`. However the group `docker` in WSL is different from the group `docker` in the container (the group id is 1001 in WSL and 999 in the container).

One solution would be to `sudo chmod 666 /var/run/docker.sock` according to this [source](https://www.digitalocean.com/community/questions/how-to-fix-docker-got-permission-denied-while-trying-to-connect-to-the-docker-daemon-socket). I'm not fond of this solution for security reasons.

A good reading to refresh memories: [Understanding how uid and gid work in Docker containers](https://medium.com/@mccode/understanding-how-uid-and-gid-work-in-docker-containers-c37a01d01cf)

A workaround to the docker groups having different gid on the host (WSL) and the container could be to user `--group-add` as mentioned here: https://github.com/moby/moby/issues/21184. Unfortunately, `group_add` for docker-compose is not longer available: https://docs.docker.com/compose/compose-file/compose-versioning/#version-2x-to-3x

**Comment by tschaffter** [comment]
Fixed in #6
